codespell_lib/_codespell.py

```diff
@@ -615,16 +633,36 @@ def main(*args):
         for word in comma_separated_words.split(','):
             ignore_words.add(word.strip())
 
-    dictionaries = options.dictionary or [default_dictionary]
-    misspellings = dict()
+    if options.dictionary:
+        dictionaries = options.dictionary
+    else:
+        dictionaries = ['-']
+    use_dictionaries = list()
     for dictionary in dictionaries:
         if dictionary == "-":
-            dictionary = default_dictionary
-        if not os.path.exists(dictionary):
-            print('ERROR: cannot find dictionary file: %s' % dictionary,
-                  file=sys.stderr)
-            parser.print_help()
-            return 1
+            # figure out which builtin dictionaries to use
+            use = sorted(set(options.builtin.split(',')))
+            for u in use:
+                for builtin in _builtin_dictionaries:
+                    if builtin[0] == u:
+                        use_dictionaries.append(
+                            os.path.join(_data_root, 'dictionary%s.txt'
+                                         % (builtin[2],)))
+                        break
+                else:
+                    print('ERROR: Unknown builtin dictionary: %s' % (u,),
+                          file=sys.stderr)
+                    parser.print_help()
+                    return 1
+        else:
+            if not os.path.isfile(dictionary):
```

Comment:
Should we flag an error here if --builtin has also been used?

Reply:
A user can validly do `--builtin clear -D - -D my_dict.txt` to use the builtin and their own. Maybe we could to catch the `--builtin clear -D my_dict` case, but even that I'm not totally convinced -- for example, someone might have a very long command (not uncommon for codespell) where they might want to temporarily remove the builtin `-D -`, and by check-and-raising we force them to also remove the `--builtin`. So there is a potential balancing act to do here. Maybe it should just be a warning? In any case I'm fine with also adding this later if need be.

Reply:
Okay, I've added a note to the project.